packages/clerk-js/src/ui/Components.tsx

```diff
@@ -149,7 +160,21 @@ const Components = (props: ComponentsProps) => {
   const { signInModal, signUpModal, userProfileModal, organizationProfileModal, createOrganizationModal, nodes } =
     state;
 
+  // TODO: rename these
+  const urlClerkState = getClerkQueryParam('__clerk_state') ?? '';
+  const decodedRedirectParams = urlClerkState ? JSON.parse(atob(urlClerkState)) : null;
```

Comment:
🔧 I'd suggest to move this to a separate utility helper e.g. `encodeBase64Json` and `decodeBase64Json`. Moreover, we must wrap this with a try/catch and cater for possible features and turn null on any error.